compiler/modules/front/stmt.bal

```diff
@@ -1158,7 +1158,17 @@ function codeGenExprForCond(StmtContext cx, bir:BasicBlock bb, Environment env,
     }
     else if flags != 0 {
         bir:Register reg = cx.createTmpRegister(t:BOOLEAN);
-        bir:AssignInsn insn = { result: reg, operand: { value, semType: t:BOOLEAN }, pos: expr.startPos };
+        bir:EqualityOp op;
+        bir:Operand equalityOperand;
+        if value {
+            equalityOperand = { value: false, semType: t:BOOLEAN };
```

Comment:
I don't understand what you are doing here. Why does it need to be false here and true in the other branch?

Shouldn't this use t:singleton?

Reply:
I think I misunderstood what you said in the [comment](https://github.com/ballerina-platform/nballerina/issues/941#issuecomment-1048373596). Fixed